app/src/main/java/com/github/se/travelpouch/model/activity/map/DirectionsViewModel.kt

```diff
@@ -41,6 +43,28 @@ class DirectionsViewModel(private val repository: DirectionsRepositoryInterface)
         }
   }
 
+  /** Function to fetch directions between activities sequentially. */
+  fun fetchDirectionsForActivities(activities: List<Activity>, mode: String, apiKey: String) {
+    if (activities.size < 2) {
+      Log.e("DirectionsViewModel", "Not enough activities to create a route")
+      return
+    }
+
+    // Extract the origin from the first activity and the destination from the last activity
+    val origin = LatLng(activities.first().location.latitude, activities.first().location.longitude)
+    val destination =
+        LatLng(activities.last().location.latitude, activities.last().location.longitude)
```

Comment:
Check if `activities.firstOrNull()?.location` and `activities.lastOrNull()?.location` are null before accessing latitude and longitude ?

Reply:
Yes nice catch